Raven.Client.Lightweight/Listeners/IDocumentSaveChangesListener.cs

```diff
@@ -0,0 +1,20 @@
+//-----------------------------------------------------------------------
+// <copyright file="IDocumentStoreListener.cs" company="Hibernating Rhinos LTD">
+//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+using Raven.Json.Linq;
+
+namespace Raven.Client.Listeners
+{
+    /// <summary>
+    /// Hook for users to provide additional logic on store operations
+    /// </summary>
+    /// 
+    public interface IDocumentSaveChangesListener
+    {
+        void BeforeSaveChanges(IDocumentSession session);
+
+        void AfterSaveChanges(IDocumentSession session);
+    }
```

Comment:
Great, now consider the case of someone who isn't doing that, and their use
case for this interface?
That is why I don't think it is required.

*Hibernating Rhinos Ltd  *

Oren Eini* l CEO l *Mobile: + 972-52-548-6969

Office: +972-4-622-7811 *l *Fax: +972-153-4-622-7811



On Tue, Apr 26, 2016 at 2:39 PM, Valeriob <notifications@github.com> wrote:

> In Raven.Client.Lightweight/Listeners/IDocumentSaveChangesListener.cs
> <https://github.com/ayende/ravendb/pull/3621#discussion_r61070544>:
>
> > +// </copyright>
> > +//-----------------------------------------------------------------------
> > +using Raven.Json.Linq;
> > +
> > +namespace Raven.Client.Listeners
> > +{
> > +    /// <summary>
> > +    /// Hook for users to provide additional logic on store operations
> > +    /// </summary>
> > +    ///
> > +    public interface IDocumentSaveChangesListener
> > +    {
> > +        void BeforeSaveChanges(IDocumentSession session);
> > +
> > +        void AfterSaveChanges(IDocumentSession session);
> > +    }
>
> I'm using an extension method on IDocumentSession.Publish(evnt) that
> stores those on session.Advanced.ExternalState["Events"] as List, and
> retrieve them on the listener via
> var events = session.Advanced.ExternalState["Events"] as List;
> https://gist.github.com/valeriob/3ae27778986da42c7a02c144d88ae52e
>
> Basically what I'm doing is customizing IDocumentSession as my UnitOfWork,
> this can be done both with wrapping the IDocumentSession or with some
> extension method and IDocumentSession.Advanced.ExternalState.
>
> I'll go forward with IDocumentSession wrapper and i'll bring this back if
> i hit any walls.
>
> Thanks for now.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/ayende/ravendb/pull/3621/files/1dcd669623231ced5d439b3caecb049a03bdf0a0#r61070544>
>